Sources/Purchasing/StoreKit2/StoreKit2TransactionListener.swift

```diff
@@ -58,7 +58,16 @@ actor StoreKit2TransactionListener: StoreKit2TransactionListenerType {
     private(set) var taskHandle: Task<Void, Never>?
 
     private weak var delegate: StoreKit2TransactionListenerDelegate?
-    private let updates: AsyncStream<TransactionResult>
+
+    // We can't directly store instances of `AsyncStream`, since that causes runtime crashes when
+    // loading this type in iOS <= 15, even with @available checks correctly in place.
+    // See https://openradar.appspot.com/radar?id=4970535809187840 / https://github.com/apple/swift/issues/58099
+    private let _updates: Any
+
+    var updates: AsyncStream<TransactionResult> {
+        // swiftlint:disable:next force_cast
+        return self._updates as! AsyncStream<TransactionResult>
+    }
```

Comment:
In this case we can use `Box` (we couldn't with `NavigationPath` because of the `@Published`).
Can you try changing this to
```suggestion
    private let _updates: Box<AsyncStream<TransactionResult>>

    var updates: AsyncStream<TransactionResult> {
        return self._updates.value
    }
```

Reply:
Done!